dts/arm/raspberrypi/rpi_pico/rp2040.dtsi

```diff
@@ -435,6 +435,34 @@
 			alarms-count = <1>;
 			status = "disabled";
 		};
+
+		sio: sio@d0000000 {
+			compatible = "raspberrypi,pico-sio";
+			reg = <0xd0000000 0x180>;
```

Comment:
Since SIO has other functions, the range of registers should be tightly fit.

See: https://github.com/zephyrproject-rtos/zephyr/pull/92923#discussion_r2249938471

Reply:
SIO is a whole block of peripherals. mbox has nothing to do with e.g. a timer or hw spinlock peripherals. So, in Zephyr convention, that would mean separate drivers and separate DT nodes.

@soburi should we maybe make SIO a node with specific peripherals as children? That could be more clear to group those together

Reply:
Don't hurry with that, as I am not sure on the idea I proposed, let's wait for @soburi to agree or disagree